thirdweb/modules/base.py

```diff
@@ -102,6 +103,26 @@ def grant_role(self, role: Role, address: str):
         )
         self.execute_tx(tx)
 
+    def upload_metadata(self, data) -> str:
+        """
+        Uploads to IPFS
+        """
+
+        if isinstance(data, str):
+            return data
+
+        if isinstance(data, dict)  :
+
+            if data.image == "":
```

Comment:
This might cause an error if the `image` property isn't set, so first you should check to see if there *is* an `image` property, like this:

```
if isinstance(data, dict) and "image" in data:
   ...
```

Also turn on your formatter (message me on discord if you have trouble with that) so that you don't have that extra two spaces before the `:` on like 114.